src/components/CompanionWindow.js

```diff
@@ -1,19 +1,91 @@
 import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import CloseIcon from '@material-ui/icons/Close';
+import IconButton from '@material-ui/core/IconButton';
+import Paper from '@material-ui/core/Paper';
+import { withStyles } from '@material-ui/core/styles';
 import ns from '../config/css-ns';
+import WindowSideBarInfoPanel from '../containers/WindowSideBarInfoPanel';
+import WindowSideBarCanvasPanel from '../containers/WindowSideBarCanvasPanel';
 
 /**
  * CompanionWindow
  */
 class CompanionWindow extends Component {
+  /**
+   * activePanelComponent
+   * @return React Component
+   */
+  activePanelComponent() {
+    const { windowId, panelContent } = this.props;
+    switch (panelContent) {
+      case 'info':
+        return <WindowSideBarInfoPanel windowId={windowId} />;
+      case 'canvas_navigation':
+        return <WindowSideBarCanvasPanel windowId={windowId} />;
+      default:
```

Comment:
This logic is now duplicated in two places.  Is there a good place to hold shared logic like this?

Reply:
I tink [that's](https://github.com/ProjectMirador/mirador/blob/master/src/components/WindowSideBarPanel.js#L15) the place jkeck refers to.